src/btree/bt_random.c

```diff
@@ -362,14 +393,16 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags)
     WT_PAGE *page;
     WT_PAGE_INDEX *pindex;
     WT_REF *current, *descent;
-    uint32_t i, entries, retry;
-    bool eviction;
+    uint32_t i, entries;
+    int retry;
```

Comment:
Here, I just had to get rid of new defined variable in later versions - `uint8_t descent_state;` and make retry a signed int.